src/libraries/Microsoft.PowerFx.Core/Parser/ParserOptions.cs

```diff
@@ -87,7 +87,8 @@ internal ParseResult Parse(string script, Features features)
             var flags = (AllowsSideEffects ? TexlParser.Flags.EnableExpressionChaining : 0) |
                         (NumberIsFloat ? TexlParser.Flags.NumberIsFloat : 0) |
                         (DisableReservedKeywords ? TexlParser.Flags.DisableReservedKeywords : 0) |
-                        (TextFirst ? TexlParser.Flags.TextFirst : 0);
+                        (TextFirst ? TexlParser.Flags.TextFirst : 0) |
+                        (AllowParseAsTypeLiteral ? TexlParser.Flags.AllowTypeLiteral : 0);
```

Comment:
>AllowParseAsTypeLiteral [](http://example.com/codeflow?start=25&length=23)

Do we need this here? I thought this is needed only to parse UserDefinitionScript (NF, UDF, TypeDefinitions).

Reply:
adding this here to introduce ParseJSON , AsType, IsType Changes later mainly . 

for now using it in ParseType function of Engine with the idea of introducing it in REPL

Reply:
I see.